src/cookie.ts

```diff
@@ -112,9 +105,20 @@ export class CookieService {
 		for (let cookieName of Object.keys(cookies)) {
 			this.delete(cookieName, path, domain);
 		}
-
 	}
 
+	/**
+	 * @private
+	 * Cookie Handler - Reusable Method
+	 *
+	 * @param  {string} name Cookie's identification
+	 * @param  {string} methodName Method to be called
+	 */
+	private cookieHandler(name: string, methodName: string): any {
+		name = encodeURIComponent(name);
+		return (new RegExp('(?:^' + name + '|;\\s*' + name + ')=(.*?)(?:;|$)', 'g') as any)
+		[methodName](document.cookie);
```

Comment:
I've created a reusable method, the line above is a substitution for the code below: 

1. check method:
`let regexp = new RegExp('(?:^' + name + '|;\\s*' + name + ')=(.*?)(?:;|$)', 'g');
 let exists = regexp.test(document.cookie);`

2. get method:
`let regexp = new RegExp('(?:^' + name + '|;\\s*' + name + ')=(.*?)(?:;|$)', 'g');
let result = regexp.exec(document.cookie);`

As you can see from the code above, both has the same regex pattern but calling different methods test & exec. I merged both patterns into a common method by passing the method name as an argument `[methodName]`.

Let me know if you have any questions.